lib/sql.js

```diff
@@ -554,13 +554,10 @@ SQLConnector.prototype.buildDelete = function(model, where, options) {
  * @param {Function} cb The callback function
  */
 SQLConnector.prototype.destroyAll = function(model, where, options, cb) {
-  var self = this;
   var stmt = this.buildDelete(model, where, options);
-  this.execute(stmt.sql, stmt.params, options, function(err, info) {
-    var affectedRows = self.getCountForAffectedRows(model, info);
-    if (cb) {
-      cb(err, {count: affectedRows});
-    }
+  this._executeAlteringQuery(model, stmt.sql, stmt.params, options, function(err, data) {
```

Comment:
Also this can be simplified a bit, with the benefit of getting rid of one unnecessary closure:

```js
this._executeAlteringQuery(model, stmt.sql, stmt.params, options, cb || function(){});
```

I usually extract the no-op callback to a top-level constant, which simplifies the code even more:

```js
// at the top:
function NOOP() {}

// here:
this._executeAlteringQuery(model, stmt.sql, stmt.params, options, cb || NOOP);
```